### DISCUSSION TITLE
Activating fprime-venv from shared github repository

### DISCUSSION THREAD

**Discussion by DrewButzel** [nasa/fprime#2674] [Q&A] [answered]
Our group has a shared GitHub repository with all our fprime code. When cloning it on to other team member's computers the fprime-venv folder isn't accessible and therefore the virtual environment can't be activated. It is in the gitignore file. Any advice on how to let all team members be able to activate the virtual environment and contribute to the code base? Should we just modify the gitignore or is there a better solution?

**Comment by thomas-bc** [comment, marked as answer]
Virtual environments should not be checked in a Git repo because they may not be compatible across platforms. 
Please see https://nasa.github.io/fprime/INSTALL.html#working-with-an-existing-project 

Basically, on any machine that wishes to collaborate on the same repo 
```
git clone --recurse-submodules <REPO>
cd <REPO>
python -m venv fprime-venv
source fprime-venv/bin/activate
pip install -r fprime/requirements.txt
```
